### DISCUSSION TITLE
I want to add my custom onnx model instead of EfficientSam.

### DISCUSSION THREAD

**Discussion by PeterKim1** [wkentaro/labelme#1397] [Q&A / Help]
Hi. I want to add my custom onnx model instead EfficientSam.

I'm working in industrial domain, so EfficientSam can't detect appropriate region in my image.

So, I need to use onnx model to detect defect regions.

What I want to do is, In labelme UI, Add 'Custom Model' category in 'AI Model' box. 

And when I set AI Model as 'Custom Model' and click 'Create AI-Polygon', then run my custom model.

I think I can modify https://github.com/labelmeai/labelme/tree/main/labelme/ai codes for my custom model,  but I don't know what parts of codes related to 'AI Model' box in labelme UI.

Could you please help me?

Many Thanks.

**Comment by wkentaro** [comment]
@PeterKim1 Yes, that's what I am aiming at as the direction of Labelme. The user will be able to use their own custom AI models. It's coming feature.

In the meantime, you need to create custom AI model as a class like `EfficientSam`:

https://github.com/labelmeai/labelme/blob/82dc203f40c3fe021adbdec3861483d57c1f6530/labelme/ai/__init__.py#L71-L84

and add to the `MODELS` list:

https://github.com/labelmeai/labelme/blob/82dc203f40c3fe021adbdec3861483d57c1f6530/labelme/ai/__init__.py#L87-L93

**Reply by Ke-Albert** [reply]
Hi. now i am using Labelme v5.9.1, does this version support custom model now? If doesn't, i can follow your suggestion to define my own model?

**Reply by wkentaro** [reply]
not yet, but the plan is the same.
